src/components/features/MyTasksView.tsx

```diff
@@ -0,0 +1,40 @@
+import { useGetTasksQuery } from "../../gql/graphql";
+import { useAppStore } from "../../store/store";
+import Error from "../errors/Error";
+import TaskColumnSkeleton from "../ui/loadingSkeletons/TaskColumnSkeleton";
+import GridView from "./GridView";
+import TableView from "./TableView";
+import { statusOptions } from "../../libs/utils";
+
+const MyTasksView = () => {
+  const viewMode = useAppStore((state) => state.viewMode);
+  const user = useAppStore((state) => state.user);
+  const searchByTaskName = useAppStore((state) => state.searchByTaskName);
+
+  const { loading, error, data } = useGetTasksQuery({
+    variables: searchByTaskName !== ""
+    ? { input: { name: searchByTaskName, assigneeId: user?.id } }
+    : {
+        input: {
+          assigneeId: user?.id,
+        },
+      },
+  });
+  
+  const tasks = data?.tasks || [];
+
+  if (loading) return <TaskColumnSkeleton />;
+  if (error) return <Error message={error.message} />;
```

Comment:
your error component receives an on retry prop, should we pass a refetch function in case something happens?